src/uagents/resolver.py

```diff
@@ -2,43 +2,71 @@
 from typing import Dict, Optional
 import random
 
-from uagents.network import get_reg_contract
+from uagents.network import get_almanac_contract, get_service_contract
 
 
-def _query_record(agent_address: str, service: str) -> dict:
-    contract = get_reg_contract()
+def query_record(agent_address: str, service: str) -> dict:
+    contract = get_almanac_contract()
     query_msg = {
         "query_record": {"agent_address": agent_address, "record_type": service}
     }
     result = contract.query(query_msg)
     return result
 
 
+def get_agent_address(name: str) -> str:
+    query_msg = {"domain_record": {"domain": f"{name}.agent"}}
+    result = get_service_contract().query(query_msg)
+    if result["record"] is not None:
+        registered_address = result["record"]["records"][0]["agent_address"]["records"]
+        if len(registered_address) > 0:
+            return registered_address[0]["address"]
+        return 0
+    return 1
+
+
+def is_agent_address(address):
+    if not isinstance(address, str):
+        return False
+
+    prefix = "agent"
+    expected_length = 65
+
+    return address.startswith(prefix) and len(address) == expected_length
+
+
 class Resolver(ABC):
     @abstractmethod
-    async def resolve(self, address: str) -> Optional[str]:
+    async def resolve(self, destination: str) -> Optional[str]:
         pass
 
 
 class AlmanacResolver(Resolver):
```

Comment:
A comment towards the naming and potential future extensions. "AlmanacResolver" supposes this resolver is only using the almanac, when in fact it is either using the almanac or the name service. Wouldn't it be more consistent to rename this one to something like "GlobalResolver" that internally calls either the AlmanacResolver or the NameServiceResolver?

Reply:
It's a good point. I guess I thought that we could handle this automatically by detecting whether the resolver is trying to resolve an address or a name. But creating an explicit NameServiceResolver as a kind of wrapper for the AlmanacResolver might indeed a nicer way to do this.

Reply:
Yeah I think its a good idea, this is down now